temporal/t.copy/t.copy.py

```diff
@@ -0,0 +1,241 @@
+
+# %option G_OPT_STDS_TYPE
+# % guidependency: input
+# % guisection: Required
+# % options: strds, str3ds, stvds
+# %end
+
+# %option G_OPT_STDS_OUTPUT
+# %end
+
+# %flag
+# % key: c
+# % label: Also copy maps of the space-time dataset
+# % description: By default the old maps are only registered in the new dataset.
+# %end
+
+
+import grass.script as gscript
+
+
+############################################################################
+
+
+def main():
+    # lazy imports
+    import grass.temporal as tgis
+
+    # Get the options
+    input = options["input"]
+    output = options["output"]
+    stdstype = options["type"]
+    copy_maps = flags["c"]
+
+    maptype = "raster"
+    element = "cell"
+    if stdstype == "str3ds":
+        maptype = "raster_3d"
+        element = "g3dcell"
+    elif stdstype == "stvds":
+        maptype = "vector"
+        element = "vector"
+
+    # Make sure the temporal database exists
+    tgis.init()
+
+    # Get the current mapset to create the id of the space time dataset
+    mapset = gscript.gisenv()["MAPSET"]
+
+    inname = input
+    inmapset = None
+    if "@" in input:
+        inname, inmapset = input.split("@")
+
+    outname = output
+    outmapset = mapset
+    if "@" in output:
+        outname, outmapset = output.split("@")
+        if outmapset != mapset:
+            gscript.fatal(
+                _("The output dataset <%s> must be in the current mapset<%s>.")
+                % (input, mapset)
+            )
+
+    # simplified version of extract_dataset() in temporal/extract.py
+    msgr = tgis.get_tgis_message_interface()
+
+    dbif = tgis.SQLDatabaseInterfaceConnection()
+    dbif.connect()
+
+    old_sp = tgis.open_old_stds(input, stdstype, dbif)
+    # Check the new stds
+    new_sp = tgis.check_new_stds(output, stdstype, dbif, gscript.overwrite())
+    if stdstype == "stvds":
+        rows = old_sp.get_registered_maps(
+            "id,name,mapset,layer", None, "start_time", dbif
+        )
+    else:
+        rows = old_sp.get_registered_maps("id,name,mapset", None, "start_time", dbif)
+
+    if rows is None or len(rows) == 0:
+        gscript.message(
+            _("Empty space-time %s dataset <%s>, nothing to copy") % (maptype, input)
+        )
+        dbif.close()
+        return
+
+    new_maps = {}
+    num_rows = len(rows)
+
+    msgr.percent(0, num_rows, 1)
+
+    if copy_maps:
+        count = 0
+
+        for row in rows:
+            count += 1
+
+            if row["mapset"] != mapset:
+                found = gscript.find_file(
+                    name=row["name"], element=element, mapset=mapset
+                )
+                if found["name"] is not None and len(found["name"]) > 0:
+                    gscript.fatal(
+                        _("A %s map <%s> exists already in the current mapset <%s>.")
+                        % (maptype, row["name"], mapset)
+                    )
+
+                kwargs = {maptype: "%s,%s" % (row["id"], row["name"])}
```

Comment:
The syntax of `g.copy` is `g.copy [raster=from,to] [raster_3d=from,to] [vector=from,to] ...`. Here I use kwargs to avoid an `if ... elif ...` construct. `maptype` is here one of raster, raster_3d, vector.